Sources/Certificate Pinning/TrustConfiguration.swift

```diff
@@ -0,0 +1,169 @@
+        ///   - includeSubdomains: Include any subdomains of the given domain in the pinning process. Defaults to true.
+        ///   - pinningHashes: An array of public key hashes that will be used to verify the presented SSL certificate.
+        ///   - expirationPolicy: The expiration policy to be used when creating this domain. A value of `.block` will cause an expired certificate to block all connections.
+        init(domain: String, enforced: Bool = true, includeSubdomains: Bool = true, pinningHashes: [Data], expirationPolicy: CertificateExpirationPolicy = .block) {
+            self.domain = domain
+            self.enforced = enforced
+            self.includeSubdomains = includeSubdomains
+            self.pinningHashes = pinningHashes
+            self.expirationPolicy = expirationPolicy
+        }
+        
+        // MARK: - Interface
+        
+        func shouldValidateCertificate(forHost host: String) -> Bool {
+            guard includeSubdomains else { return domain == host }
+            return host.contains(domain)
+        }
+        
+        func shouldValidateCertificate(forHost host: String, at date: Date) -> Bool {
+            guard shouldValidateCertificate(forHost: host), let expired = expirationPolicy.expiration else { return true }
+            return date < expired
+        }
+        
+        func validate(against remoteCertificate: SecCertificate) -> Bool {
+            let pinningHash = try? CertificateHasher.pinningHash(for: remoteCertificate)
+            return pinningHash.map(pinningHashes.contains) ?? false
+        }
+        
+        var dispositionForFailedValidation: URLSession.AuthChallengeDisposition {
+            if enforced {
+                return .cancelAuthenticationChallenge
+            }
+            
+            debugPrint("SSL certificate pin failed for host: '\(domain)'. Configuration set to allow connection anyway.")
+            return .performDefaultHandling
+        }
+    }
+    
+    // MARK: - Properties
+    
+    /// A list of `DomainConfiguration` objects which each represent a single domain's SSL pinning configuration.
+    public let domainConfigurations: [DomainConfiguration]
+    
+    // MARK: - Initializers
+    
+    /// Create a `TrustConfiguration` instance which will govern how the validator behaves when presented with a remote SSL certificate.
+    ///
+    /// - Parameter domainConfigurations: A list of `DomainConfiguration` objects which control the pinning process for each individual domain.
+    ///         There must only be a single domain configuration for a given domain. Multiple will trigger an assertion.
+    public init(domainConfigurations: [DomainConfiguration]) {
+        assert(Set(domainConfigurations.map { $0.domain }).count == domainConfigurations.count, "You must not provided multiple domain configurations for any given domain.")
```

Comment:
Typo:
You must not provided multiple domain configurations -> You must not provide multiple domain configurations